src/quantity/_src/array_api.py

```diff
@@ -0,0 +1,24 @@
+"""Minimal definition of the Array API.
+
+NOTE: this module will be deprecated when
+https://github.com/data-apis/array-api-typing is released.
+
+"""
+
+from __future__ import annotations
+
+__all__ = ["HasArrayNameSpace", "Array"]
+
+from typing import Any, Protocol
+
+
+class HasArrayNameSpace(Protocol):
+    """Minimal definition of the Array API."""
+
+    def __array_namespace__(self) -> Any: ...
+
+
+class Array(HasArrayNameSpace, Protocol):
+    """Minimal definition of the Array API."""
+
+    def __pow__(self, other: Any) -> Array: ...
```

Comment:
I think mypyc has fixed the `Self` annotation problem. Should probably change this to `Self`, then work on getting mypyc set up.

Reply:
Do we actually need `__pow__`? `__array_namespace__` should be enough, no? If we do think more is needed, I'd prefer to add something like `shape` and `dtype`.

Reply:
Ideally `Array` would define the full gamut from https://data-apis.org/array-api/latest/API_specification/array_object.html. Otherwise static typers wouldn't know that you can e.g. add 2 quantities together.

Reply:
I had `__pow__` because that's all that was needed at the time I started PR #8.

Reply:
Not sure I fully understand, but the top is pretty clear that we hope the array API folds themselves will give us this eventually, so should be fine.

Reply:
> Should probably change this to Self, then work on getting mypyc set up.

I'd actually rather defer and see what happens with mypyc.